pkg/app/piped/platformprovider/ecs/routing_traffic.go

```diff
@@ -14,9 +14,37 @@
 
 package ecs
 
+import (
+	"sort"
+
+	"github.com/aws/aws-sdk-go-v2/service/elasticloadbalancingv2/types"
+)
+
 type RoutingTrafficConfig []targetGroupWeight
 
 type targetGroupWeight struct {
 	TargetGroupArn string
 	Weight         int
 }
+
+func (c RoutingTrafficConfig) hasSameTargets(forwardActionTargets []types.TargetGroupTuple) bool {
+	if len(c) != len(forwardActionTargets) {
+		return false
+	}
+
+	// Sort so that the both slices have the same target groups.
+	sort.Slice(c, func(i, j int) bool {
+		return c[i].TargetGroupArn < c[j].TargetGroupArn
+	})
+	sort.Slice(forwardActionTargets, func(i, j int) bool {
+		return *forwardActionTargets[i].TargetGroupArn < *forwardActionTargets[j].TargetGroupArn
+	})
+
+	for i := range c {
+		if c[i].TargetGroupArn != *forwardActionTargets[i].TargetGroupArn {
+			return false
+		}
+	}
```

Comment:
Instead of sorting both, then iterating through, using an extra map with TargetGroupArn as key could provide a better performance. It's a more straightforward approach.

Reply:
@khanhtc1202 Thanks! I'll try

Reply:
@khanhtc1202 
I fixed on https://github.com/pipe-cd/pipecd/pull/4733/commits/727e13c6e407793ee28a8bff8d5ec5e11728c482